### ISSUE TITLE
Cuda availability

### ISSUE THREAD

**Issue by Surbhi-xerxes** [nickkunz/smogn#23]
Hi
Thanks for the amazing library. However, the number of data samples I have, is quite large. For now, it's going to take days to run this computation. Is it possible to run this code on google colab's gpu? Is your code automated to run on cuda?

**Comment by ashikshafi08** [comment]
Hey @Surbhi-xerxes were you able to run this on CUDA? I'm trying to work with a similar dataset as well (pretty huge) the iterations taking forever.

**Comment by Surbhi-xerxes** [comment]
No. I think the way it is implemented, it cannot be parallelized and that is why, we dont have a cuda implementation for the same. Since I was dealing with a regression problem, I , instead, divided my dataset into bins and assigned class weights to these bins. I then oversampled/undersampled based on the class weights. This did not remove the class imbalance for me though, but atleast I could undersample based on the class weights.
